scheduler/task.go

```diff
@@ -537,6 +535,37 @@ func (t *task) fire() {
 	t.state = core.TaskSpinning
 }
 
+// disable proceeds disabling a task which consists of changing task state to disabled, emitting an appropriate event
+// and unsubscribing its dependencies
+func (t *task) disable(failureMsg string) {
+	t.Lock()
+	t.state = core.TaskDisabled
+	t.Unlock()
+
+	// Send task disabled event
+	event := new(scheduler_event.TaskDisabledEvent)
+	event.TaskID = t.id
+	event.Why = fmt.Sprintf("Task disabled with error: %s", failureMsg)
+	defer t.eventEmitter.Emit(event)
+
+	// We need to unsubscribe from deps when a task has disabled
+	t.UnsubscribePlugins()
```

Comment:
Answer to 
> Only concern is that the disabled task can be enabled. Do we need to subscribe it before enabling it?
The disabled task can be enabled and yes, we need to subscribe it before enabling it (exactly it is a part of enabling procedure). To be clear this PR does not impact on such behavior. 
> adding error handling
There was a discussion about what should happen first - emitting an event or unsubscribe deps. The decision is to keep it as it is what means emitting event as first, and then do unsubscribing in HandleGomitEvent. I will revert my changes to this aspect.